lab_1/main.py

```diff
@@ -116,16 +188,43 @@ def detect_language_advanced(unknown_profile: dict, profiles: list, languages: l
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
-
+    if not(isinstance(unknown_profile, dict) and isinstance(profiles, list)
+           and isinstance(languages, list) and isinstance(top_n, int)):
+        return None
+    reports = []
+    for current_profile in profiles:
+        if current_profile['name'] in languages or not languages:
+            reports.append(compare_profiles_advanced(unknown_profile, current_profile, top_n))
+    if not reports:
+        return None
+    reports = sorted(reports, key=lambda k: k['score'], reverse=True)
+    # to put languages in alphabetical score if there are some similar scores
+    counter = 0
+    for i in range(len(reports)-1):
+        if reports[i]['score'] == reports[i+1]['score']:
+            counter += 1
+        else:
+            break
+    if counter != 0:
+        language = sorted([reports[i]['name'] for i in range(0, counter+1)])[0]
```

Comment:
can you explain this line, please?

Reply:
I introduced variable "counter" to find out if there are some languages with similar scores. If counter != 0, i. e. we have such languages, we generate the list ant put there the names of these languages. After that, we put the languages in alphabetical score and take the first one. That's the language we should return.